File: .kotlin_alpha/services/ec2/src/main/kotlin/com/kotlin/ec2/DescribeInstances.kt
//snippet-sourcedescription:[DescribeInstances.kt demonstrates how to get information about all the Amazon Elastic Compute Cloud (Amazon EC2) Instances associated with an AWS account.]
//snippet-keyword:[AWS SDK for Kotlin]
//snippet-keyword:[Code Sample]
//snippet-service:[Amazon EC2]
//snippet-sourcetype:[full-example]
//snippet-sourcedate:[07/21/2021]
//snippet-sourceauthor:[scmacdon-aws]

/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.ec2

// snippet-start:[ec2.kotlin.describe_instances.import]
import aws.sdk.kotlin.services.ec2.Ec2Client
import aws.sdk.kotlin.services.ec2.model.DescribeInstancesRequest
import aws.sdk.kotlin.services.ec2.model.Ec2Exception
import kotlin.system.exitProcess
// snippet-end:[ec2.kotlin.describe_instances.import]

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun main() {

    val ec2Client = Ec2Client{region = "us-west-2"}
    describeEC2Instances(ec2Client)
}

// snippet-start:[ec2.kotlin.describe_instances.main]
suspend fun describeEC2Instances(ec2: Ec2Client) {

    try {

        val request = DescribeInstancesRequest{
                maxResults = 6
            }

        val response = ec2.describeInstances(request)
            for (reservation in response.reservations!!) {
                for (instance in reservation.instances!!) {
                    println("Instance Id is ${instance.instanceId}")
                    println("Image id is ${instance.imageId}")
                    println("Instance type is ${instance.instanceType}")
                    println("Instance state name is ${instance.state?.name}")
                    println("monitoring information is ${instance.monitoring?.state}")
                }
            }

    } catch (e: Ec2Exception) {
        println(e.message)
        exitProcess(0)
    }
}
// snippet-end:[ec2.kotlin.describe_instances.main]

File: php/example_code/ec2/DescribeInstances.php
<?php
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
/*
 *  ABOUT THIS PHP SAMPLE: This sample is part of the SDK for PHP Developer Guide topic at
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/ec2-examples-using-elastic-ip-addresses.html
 *
 *
 *
 */

// snippet-start:[ec2.php.describe_instances.complete]
// snippet-start:[ec2.php.describe_instances.import]

require 'vendor/autoload.php';

use Aws\Ec2\Ec2Client;

// snippet-end:[ec2.php.describe_instances.import]
/**
 * Describe Instances
 *
 * This code expects that you have AWS credentials set up per:
 * https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_credentials.html
 */

// snippet-start:[ec2.php.describe_instances.main]
$ec2Client = new Aws\Ec2\Ec2Client([
    'region' => 'us-west-2',
    'version' => '2016-11-15',
    'profile' => 'default'
]);
$result = $ec2Client->describeInstances();
echo "Instances: \n";
foreach ($result['Reservations'] as $reservation) {
    foreach ($reservation['Instances'] as $instance) {
        echo "InstanceId: {$instance['InstanceId']} - {$instance['State']['Name']} \n";
    }
}

// snippet-end:[ec2.php.describe_instances.main]
// snippet-end:[ec2.php.describe_instances.complete]
